sql/handler.cc

```diff
@@ -7225,6 +7225,13 @@ bool Vers_parse_info::fix_alter_info(THD *thd, Alter_info *alter_info,
   if (!need_check(alter_info) && !share->versioned)
     return false;
 
+  if (alter_info->flags & Alter_info::ALTER_ADD_SYSTEM_VERSIONING &&
```

Comment:
Is it really needed to check `ALTER_ADD_SYSTEM_VERSIONING` flag?

Reply:
Yes. Or server will show incorrect warnings in cases like this:
```sql
alter table t1 DROP system versioning' failed: 4145: ADD SYSTEM VERSIONING prohibited for TEMPORARY tables
```

Reply:
Then check `share->tmp_table`  instead of `share->table_category`.

Reply:
Done.

Reply:
The topic was about removing `ALTER_ADD_SYSTEM_VERSIONING` check.

Reply:
Updated. Now `ALTER_ADD_SYSTEM_VERSIONING` removed.